lapce-ui/src/editor/tab_header_content.rs

```diff
@@ -248,6 +249,79 @@ impl LapceEditorTabHeaderContent {
             ));
         }
     }
+
+    fn get_effective_path(
+        &mut self,
+        workspace_path: Option<PathBuf>,
+        file_path: &PathBuf,
+    ) -> String {
+        let workspace_path = match workspace_path {
+            Some(p) => p,
+            None => {
+                // file_path.truncate(20);
+                let file_path = file_path
+                    .to_string_lossy()
+                    .char_indices()
+                    .filter(|(i, _)| *i < 20)
+                    .map(|(_, ch)| ch)
+                    .collect::<String>();
+                return file_path;
+            }
+        };
+
+        // file is a workspace file: we can truncate to workspace folders
+        if file_path.starts_with(&workspace_path) {
```

Comment:
I haven't given this much thought, really, but if you want to trim the workspace prefix, you can do that with `str`'s [`strip_prefix`](https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix). I think the problems panel implementation does this already if you want an example.

Reply:
I do want to trim the workspace prefix (make it so that the path shown will be relative inside the workspace). Nevertheless, using `strip_prefix` involves some normalization issues (trailing slashes when popping the file, etc.), so I would say using pathbuff would be the easiest way to handle this case. There might be a more efficient way though, and I will try to look into it